reconcile/utils/state.py

```diff
@@ -34,20 +41,44 @@ def __init__(self, integration: str, accounts: Iterable[Mapping[str, Any]],
 
         self.client = session.client('s3')
 
+        # check if the bucket exists
+        try:
+            self.client.head_bucket(Bucket=self.bucket)
+        except ClientError as details:
+            error_code = details.response.get('Error', {}).get('Code', None)
+            if error_code == '404':
```

Comment:
i am seeing cases where a 403 is returned. do we care about error_code here? bucket is just inaccessible...

Reply:
you can add `details` to the exception message you are raising here?